pipeline/backend/kubernetes/service.go

```diff
@@ -24,6 +26,7 @@ func Service(namespace, name string, ports []uint16) (*v1.Service, error) {
 	var svcPorts []v1.ServicePort
 	for _, port := range ports {
 		svcPorts = append(svcPorts, v1.ServicePort{
+			Name:       strconv.FormatUint(uint64(port), 10),
```

Comment:
maybe we could name them `port-X` - `fmt.Sprintf("port-%d", port)`?

Reply:
I think we could, but `strconv` seems a way faster, if my benchmark is correct 

```go
package main

import (
        "fmt"
        "time"
        "strconv"
)

func printf(num uint16) time.Duration {
        start := time.Now()
        _ = fmt.Sprintf("port-%d", num)
        return time.Since(start)
}

func convert(num uint16) time.Duration {
        start := time.Now()
        _ = strconv.FormatUint(uint64(num), 10)
        return time.Since(start)
}
func main(){
        var num uint16 = 1000

        fmt.Printf("%v - %v\n\n", convert(num).Nanoseconds(), printf(num).Nanoseconds())
        fmt.Printf("%v - %v\n\n", convert(num).Nanoseconds(), printf(num).Nanoseconds())
        fmt.Printf("%v - %v\n\n", convert(num).Nanoseconds(), printf(num).Nanoseconds())
        fmt.Printf("%v - %v\n\n", convert(num).Nanoseconds(), printf(num).Nanoseconds())
        fmt.Printf("%v - %v\n\n", convert(num).Nanoseconds(), printf(num).Nanoseconds())
        fmt.Printf("%v - %v\n\n", convert(num).Nanoseconds(), printf(num).Nanoseconds())
        fmt.Printf("%v - %v\n\n", convert(num).Nanoseconds(), printf(num).Nanoseconds())
}
```

```bash 
go run main.go

438 - 12970

157 - 327

96 - 174

92 - 161

63 - 141

64 - 152

46 - 167

```

Ofc it's just nanoseconds, but anyway, why making it slower

Reply:
Yes, of course specific convert function will be faster than generic format but this was more about readability just in case you need to debug it in k8s cluster etc so that's it's easy to identify that it's port name

Reply:
Not sure how it's going to help, because in code it's under the `Name` element, and if you get a service from k8s, it's going to look like 
```yaml
apiVersion: v1
kind: Service
metadata:
  name: wp-01hhf980hdzs25ewwe2am6g3ad-0-services-0
  namespace: woodpecker-ci
spec:
  clusterIP: 10.152.183.105
  clusterIPs:
  - 10.152.183.105
  internalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - name: "2375"
    port: 2375
    protocol: TCP
    targetPort: 2375
  - name: "16443"
    port: 16443
    protocol: TCP
    targetPort: 16443
  selector:
    step: wp-01hhf980hdzs25ewwe2am6g3ad-0-services-0
  sessionAffinity: None
  type: ClusterIP
```

So it's like pretty obvious that it's name and nothing else.

But if you'd like to have it with the `port` prefix anyway, I don't mind changing, just don't think it's making it more readable.

Reply:
I think, nevermind, I've just updated to `sptrinf`

Reply:
At current point it probably does not matter but I have not tested how it does behave in SRV record nslookup etc when name is just numeric just in case we would use it at some point